### PR TITLE
Expand plans API

### PR DESCRIPTION
**Pull request by hatched**
Add in the missing plans API so they can be used in the subsequent deployment flow UI updates.

### CODE REVIEW

**Comment by hatched** [comment]
No QA necessary as this will be exercised in following branches.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5546/
Test PASSed.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/index.js [hunk 1]
```diff
@@ -70,6 +70,9 @@ var module = module;
       case 'PUT':
         return bakery.sendPutRequest(
             path, JSON.stringify(params), success, failure, redirect);
+      case 'PATCH':
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/index.js]: Nice, thanks!

**Comment by huwshimi** [comment]
:+1:

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/plans.js [hunk 2]
```diff
@@ -172,6 +172,92 @@ var module = module;
     },
 
     /**
+      Changes the budget to the new limit.
+
+      @method updateBudget
```
Comment (frankban) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/plans.js]: I know this is already merged, but hope this comment can still be useful. I think we should always create a customized, JS friendly, result value from the API server responses. In essence, every time we just `return jujulib._makeRequest(...` we are propagating the server side results to the rest of the application. If the response changes (we know it shouldn't right?) then the application is broken and the error is not intercepted here, but somewhere internally. Moreover, by looking at this code I have basically no idea about what's this `data` passed to the callbacks. I think we should make our assumptions about server side responses more explicit and clear, also considering that we are forcefully using mocks in the tests. For an example of how to do this, see _handlePlan() in this same file.

**Pull request merged**